plugins/quota/actor.go

```diff
@@ -0,0 +1,105 @@
+package quota
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	gderrors "github.com/gluster/glusterd2/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type quotadActor struct{}
+
+const (
+	quotaDaemonKey = "enable"
+)
+
+// IsQuotadAffectedNode returns true if there are local bricks of volume on
+// which quota is enabled
+func IsQuotadAffectedNode() bool {
+	volumes, e := volume.GetVolumes()
+	if e != nil {
+		log.WithError(e).Error("Failed to get volumes")
+		return true
+	}
+	for _, v := range volumes {
+		val, exists := v.Options[quotaEnabledKey]
+		if exists && val == "off" {
+			continue
+		} else if v.State != volume.VolStarted {
+			continue
+		} else {
+			bricks := v.GetLocalBricks()
+			if len(bricks) > 0 {
+				return true
+			}
+			continue
+		}
+	}
+	return false
+}
+
+func (actor *quotadActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
+	if key != quotaDaemonKey {
+		return nil
+	}
+	quotadDaemon, err := NewQuotad()
+	if err != nil {
+		return err
+	}
+
+	switch value {
+	case "on":
+
+		// Create pidfile dir if not exists
+		if err := os.MkdirAll(path.Dir(quotadDaemon.pidfilepath),
+			os.ModeDir|os.ModePerm); err != nil {
+			return err
+		}
+
+		// Create logFiledir dir
+		if err := os.MkdirAll(path.Dir(quotadDaemon.logfilepath),
+			os.ModeDir|os.ModePerm); err != nil {
+			return err
+		}
+
+		err = daemon.Stop(quotadDaemon, true, logger)
+		if err != nil && err != gderrors.ErrProcessNotRunning && err != gderrors.ErrPidFileNotFound {
+			return err
+		}
+		err = daemon.Start(quotadDaemon, true, logger)
+		if err != gderrors.ErrProcessAlreadyRunning || err != nil {
```

Comment:
Shouldn't it be `&&` here ?

Also, check for `nil` first

Reply:
this had a case missing. have fixed it.